cuqi/experimental/mcmc/_sampler.py

```diff
@@ -13,6 +13,27 @@ def progressbar(iterable, **kwargs):
         warnings.warn("Module mcmc: Progressbar not found. Install progressbar2 to get sampling progress.")
         return iterable
 
+class _UniqueList(list):
```

Comment:
Is there a particular need for this data structure, instead of using Python set. for example:
```
set1 = {'key1'} # set elements are 'key1'
set1.add('key1') #  set elements are 'key1'
set1.add('key2') #  set elements are 'key1', 'key2'
set1.update({'key2', 'key3'})  #  set elements are 'key1', 'key2', 'key3'
```

Reply:
done